src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,97 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao;
+    private static RegistrationServiceImpl registrationService;
+    private static User[] users;
+
+    @BeforeAll
+    static void beforeAll() {
+        final int usersNumbs = 7;
+        registrationService = new RegistrationServiceImpl();
+        users = new User[usersNumbs];
+        storageDao = new StorageDaoImpl();
+        users[0] = new User("pokemon", "Rock99pop", 23);
+        users[1] = new User("donald", "Tot77", 18);
+        users[2] = new User("supra", "Lol556yo", 16);
+        users[3] = new User("superCat", null, 23);
+        users[4] = new User(null, "Toyota45cut", 23);
+        users[5] = new User("logoBulls", "Toronto78punk", -53);
+        users[6] = new User("", "", 23);
+    }
+
+    @Test
+    void addTest_OK() throws RegistrationException {
+        final int numbers = 1;
+        registrationService.register(users[0]);
+        boolean expected = true;
+        boolean actual = Storage.people.size() == numbers;
+        assertEquals(expected, actual, "There must be one since one user has been added!");
+    }
+
+    @Test
+    void overwritingNotPossible_notOK() throws RegistrationException {
```

Comment:
follow this pattern 
<img width="800" alt="image" src="https://user-images.githubusercontent.com/81781479/221670752-e841dc9c-97c2-4551-a037-55989dcaa40b.png">